### PR TITLE
Convert ignored local objects from error to warning

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/KAC-279

Changes:
- Unsupported/ignored component in the local manifest generates warning not error (for example a sandbox configuration from the past).

### CODE REVIEW

File: internal/pkg/model/filter.go [hunk 1]
```diff
@@ -16,6 +17,15 @@ const (
 	MainBranchDef  = "__main__"
 )
 
+const (
+	IgnoredByAllowedKeys IgnoreReason = iota
+	IgnoredByAllowedBranches
+	IgnoredByIgnoredComponents
+	IgnoredByAlwaysIgnoredComponents
+)
```
Comment (michaljurecko) [inline internal/pkg/model/filter.go:25]: Do chyby, ze nieco ignorujeme pridavam dovod, aby sme mohli vygenerovat lepsiu spravu.

Comment (michaljurecko) [reply internal/pkg/model/filter.go:25]: (moze byt, alebo niekoho napada lepsi naming?)

File: internal/pkg/model/filter.go [hunk 2]
```diff
@@ -53,24 +68,45 @@ func NoFilter() ObjectsFilter {
 }
 
 func (f ObjectsFilter) IsObjectIgnored(object Object) bool {
+	return f.AssertObjectAllowed(object) != nil
+}
+
+func (f ObjectsFilter) AssertObjectAllowed(object Object) *ObjectIsIgnoredError {
```
Comment (michaljurecko) [inline internal/pkg/model/filter.go:74]: `IsObjectIgnored` -> `AssertObjectAllowed`

File: internal/pkg/state/load_local_test.go [hunk 3]
```diff
@@ -105,24 +106,33 @@ func TestLoadLocalStateAllowedBranches(t *testing.T) {
...
+	assert.NoError(t, localErr)
+
+	expected := `
+DEBUG  Loading local state.
+WARN  found manifest record for branch "123":
+- it is not allowed by the manifest definition
+- please, remove record from the manifest and the related directory
+- or modify "allowedBranches" key in the manifest
+`
+	assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(d.DebugLogger().AllMessages()))
```
Comment (michaljurecko) [inline internal/pkg/state/load_local_test.go:129]: Uz to nie je error ale warning.

File: internal/pkg/state/local/manager.go [hunk 4]
```diff
@@ -165,11 +165,32 @@ func (u *UnitOfWork) LoadObject(manifest model.ObjectManifest, filter model.Obje
...
 			// Validate, object must be allowed
-			if filter.IsObjectIgnored(object) {
-				return errors.Errorf(
-					`found manifest record for %s "%s", but it is not allowed by the manifest definition`,
-					object.Kind().Name,
-					object.ObjectId())
+			if err := filter.AssertObjectAllowed(object); err != nil {
+				switch err.Reason() {
+				case model.IgnoredByAllowedBranches:
+					u.logger.Warn(errors.NewNestedError(
```
Comment (michaljurecko) [inline internal/pkg/state/local/manager.go:171]: Implementacia.

**Review by JakubMatejka — APPROVED** [review]

**Pull request merged**